src/compile.rs

```diff
@@ -682,6 +706,145 @@ fn check_required_config<'a>(
     ret.into_result(())
 }
 
+pub fn get_infos<'a>(
```

Comment:
This function seems kind of weird to me.  It looks like it does all the compilation for an individual system except for types.  Which is a slightly strange distinction to make.  I think I'm struggling to interpret and understand this function because the intent is not clear to me.  Perhaps with a more descriptive function name and a comment explaining what this function does it will make sense?  I'm more inclined to think that it does several mostly unrelated things and should either be split up, or it should include the entirety of system building (including types).